<launch>
    
    <arg name="launch_prefix" default="" /> <!-- set to value="gdbserver localhost:10000" for remote debugging -->
    <arg name="node_namespace" default="apriltag_ros_continuous_node" />
    <arg name="camera_name" default="mjpeg_cam" />
    <arg name="camera_frame" default="/usb_cam" />
    <arg name="image_topic" default="image" />

    <!-- Set parameters -->
    <rosparam command="load" file="$(find state_estimator)/config/settings_compressed.yaml" ns="$(arg node_namespace)" />
    <rosparam command="load" file="$(find state_estimator)/config/tags.yaml" ns="$(arg node_namespace)" />

    <node name="mjpeg_cam" pkg="mjpeg_cam" type="mjpeg_cam" output="screen" >
        <param name="device_name" value="/dev/video0" />
        <param name="width" value="640" />
        <param name="height" value="480" />
        <param name="framerate" value="30" />
        <param name="exposure" value="70" />
        <param name="brightness" value="128" />
        <param name="autoexposure" value="false" />
    </node>
    <node pkg="state_estimator" type="CompressedToImageRaw.py" name="image_conversion"/>
    <node pkg="state_estimator" type="TagToNavMsg.py" name="usb_cam_tag_conversion" />

    <node pkg="apriltag_ros" type="apriltag_ros_continuous_node" name="$(arg node_namespace)" clear_params="true" output="screen" launch-prefix="$(arg launch_prefix)" >
        <!-- Remap topics from those used in code to those on the ROS network -->
        <remap from="image_rect" to="$(arg camera_name)/$(arg image_topic)" />
        <remap from="camera_info" to="$(arg camera_name)/camera_info" />
        <param name="camera_frame" type="str" value="$(arg camera_frame)" />
        <param name="publish_tag_detections_image" type="bool" value="true" />      <!-- default: false -->
    </node>

</launch>
